AutomatedTesting/Gem/PythonTests/Prefab/tests/reparent_prefab/ReparentPrefab_UnderPrefabAndEntityHierarchies.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+def ReparentPrefab_UnderPrefabAndEntityHierarchies():
+
+    from pathlib import Path
+    CAR_PREFAB_FILE_NAME = Path(__file__).stem + '_car_prefab'
+    WHEEL_PREFAB_FILE_NAME = Path(__file__).stem + '_wheel_prefab'
+    DRIVER_PREFAB_FILE_NAME = Path(__file__).stem + '_driver_prefab'
+
+    import editor_python_test_tools.pyside_utils as pyside_utils
+
+    @pyside_utils.wrap_async
+    async def run_test():
+
+        from editor_python_test_tools.editor_entity_utils import EditorEntity, EditorLevelEntity
+        from editor_python_test_tools.prefab_utils import Prefab, wait_for_propagation
+
+        import Prefab.tests.PrefabTestUtils as prefab_test_utils
+        import azlmbr.legacy.general as general
+
+        async def reparent_with_undo_redo(prefab_instance, new_parent_entity_id):
+            # Get child data on the original parent, and reparent to new entity
+            original_parent = EditorEntity(prefab_instance.container_entity.get_parent_id())
+            original_parent_original_children_ids = {child_id.ToString(): child_id for child_id in
+                                                     original_parent.get_children_ids()}
+            await prefab_instance.ui_reparent_prefab_instance(new_parent_entity_id)
+
+            # Undo the reparent operation, and verify original parent is restored
+            general.undo()
+            wait_for_propagation()
+            original_parent_new_children_ids = {child_id.ToString(): child_id for child_id in
+                                                original_parent.get_children_ids()}
+            assert original_parent_original_children_ids == original_parent_new_children_ids, \
+                "Failed to reparent instance to the original parent via Undo"
+
+            # Redo the reparent operation, and verify the new instance is not among the original parent's child entities
+            general.redo()
+            wait_for_propagation()
+            original_parent_new_children_ids = {child_id.ToString(): child_id for child_id in
+                                                original_parent.get_children_ids()}
+            assert prefab_instance.container_entity.id not in original_parent_new_children_ids, \
+                "Redo on reparent operation failed"
+
+        prefab_test_utils.open_base_tests_level()
+
+        Report.info(f"Level Entity Id: {EditorEntity(EditorLevelEntity).id}")
```

Comment:
Doesn't seem like its used anywhere else and doesn't seem to add much value to the test. Can be removed?

Reply:
Good catch, this was a debugging addition I forgot to remove.